repo.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+// DiffCommitBranch commits ahead and commits behind for a given branch and base branch
+type DiffCommitBranch struct {
+	branch     string
+	baseBranch string
+	ahead      []*object.Commit
+	behind     []*object.Commit
+}
+
+func loadOrClone(repoURL string, directory string, remoteName string, skipFetch bool) (*git.Repository, error) {
+	var repo *git.Repository
+	var err error
+
+	if directory == "" {
+		return nil, errors.New("no directory provided")
+	}
+
```

Comment:
check repoURL == null and return error

Reply:
Providing a repoURL will work by cloning on an empty directory. Not providing it, it works with an existing git repo, so it's not an error.